os/pm/pm_activity.c

```diff
@@ -130,6 +132,24 @@ void pm_activity(int domain, int priority)
 
 		pdom->accum = (int16_t)accum;
 
+		#ifdef CONFIG_PM_METRICS
+			FAR sq_entry_t *node = NULL;
+			int flag = ERROR;
+			for (node = sq_peek(&(pdom->accum_history)); node; node = node->flink) {
+				if (!strcmp(((struct pm_accumchange_s*)node)->name, callbacks->name)) {
```

Comment:
You'd better use `strncmp` instead of `strcmp`.

Reply:
modified.